resource/plugins/unreal/python/loader/finalizers/unreal_asset_loader_finalizer.py

```diff
@@ -0,0 +1,171 @@
+                ).one()
+                dependency_ident = str_version(
+                    dependency_assetversion, by_task=False
+                )
+                self.logger.debug(
+                    'Restoring pipeline asset info for {}: {}'.format(
+                        dependency_ident, asset_info
+                    )
+                )
+                ftrack_object_manager = UnrealFtrackObjectManager(
+                    self.event_manager
+                )
+                ftrack_object_manager.asset_info = asset_info
+                dcc_object = UnrealDccObject()
+                dcc_object.name = (
+                    ftrack_object_manager._generate_dcc_object_name()
+                )
+                if dcc_object.exists():
+                    self.logger.debug(
+                        'Pipeline asset {} already tracked in Unreal, removing!'.format(
+                            dependency_ident
+                        )
+                    )
+                    unreal_utils.delete_ftrack_node(dcc_object.name)
+                # Store asset info
+                dcc_object.create(dcc_object.name)
+                # Have it sync to disk
+                ftrack_object_manager.dcc_object = dcc_object
+            else:
+                # Remove metadata
+                if unreal_utils.conditional_remove_metadata_tag(
+                    asset_path, asset_const.NODE_METADATA_TAG
+                ):
+                    self.logger.debug(
+                        'Removed pipeline asset metadata tag from asset: {}'.format(
+                            asset_path
+                        )
+                    )
+        # Remove snapshot metadata tag carried along - this is not a dependency (yet)
+        if unreal_utils.conditional_remove_metadata_tag(
+            asset_path, asset_const.NODE_SNAPSHOT_METADATA_TAG
+        ):
+            self.logger.debug(
+                'Removed snapshot asset metadata tag from asset: {}'.format(
+                    asset_path
+                )
+            )
+
+        # Restore file modification time if possible
+        if metadata is not None:
+            if asset_const.FILE_SIZE in metadata:
+                file_size = metadata[asset_const.FILE_SIZE]
+                imported_file_size = os.path.getsize(asset_filesystem_path)
+                mode_date = metadata[asset_const.MOD_DATE]
+                if file_size == imported_file_size:
+                    # Same size, lets assume it is the same file
+                    stat = os.stat(asset_filesystem_path)
+                    os.utime(
+                        asset_filesystem_path, times=(stat.st_atime, mode_date)
+                    )
+                    self.logger.debug(
+                        'Restored file modification time: {} on asset: {} (size: {})'.format(
+                            mode_date, asset_path, file_size
+                        )
+                    )
+                else:
```

Comment:
This should be a utils function

Reply:
Removed the restore of modification date as discussed.